ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -260,6 +260,7 @@ mkNodeArgs registry cfg initState tracers btime chainDB = NodeArgs
     , blockFetchSize      = nodeBlockFetchSize
     , blockMatchesHeader  = nodeBlockMatchesHeader
     , maxUnackTxs         = 100 -- TODO
+    , maxBlockSize        = 2000000 -- TODO
```

Comment:
@dcoutts Should this come from the genesis config too?

Also, how is this defined? I'm assuming it is the size of the block *body*, not the size of the whole block, including the header.

Reply:
Perhaps we should instead call this `maxBlockBodySize`, then?

Also, this value probably wouldn't directly translate to the transactions we can select from the mempool by summed byte size. The block `Body` would consist of other extra bytes from the CBOR encodings of all the payloads (`TxPayload`, `Delegation.Payload`, `Update.Payload`).

Reply:
Name is fine for me. We could indeed account for these extra bytes because of CBOR tags and `listLen` bytes (I believe the number of bytes is static). As you suggested on Slack, doing that in `cardano-ledger` is less fragile. @dcoutts What do you think?

Reply:
The name is fine. The max block size actually comes from the current ledger protocol params (with the initial value in the genesis).

The block size refers to the total block size, including the block header and all other encoding overheads.

I suggest that we approximate this by a per-ledger-instance constant max overhead, so we would end up taking `maxBlockSize - blockEncodingOverheadsSize` worth of txs.